### DISCUSSION TITLE
18 gennaio 2022 B - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#6] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/8fdac7ab-e7a6-4354-a2f9-e37fcf21ff3d)

**Comment by niccolo-fato** [comment]
![2024-01-13 11 00 55](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/a929be38-8569-4e99-bd4a-33d4a72c3090)
![2024-01-13 11 01 04](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/b5de5726-a53b-4a18-8ded-462b8da79fef)
![2024-01-13 11 01 12](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/f02ff4ca-e118-4dda-afd8-f7e2fb3d7360)

**Reply by Alegau03** [reply]
Non capisco perchè toglie BE->A, fino alla decomposizione la tieni e dopo la cancelli

**Reply by niccolo-fato** [reply]
Perché dentro la decomposizione ho già AEB

**Reply by IonutCicio** [reply]
La toglie perchè il sottoschema generato da $AE \to B \Rightarrow AEB = BEA \Leftarrow BE \to A$, generano lo stesso sottoschema quindi lo può anche levare. Se non sei convinto, basta notare che quando scriviamo $AEB$ intendiamo l'insieme $\set{A, E, B}$, e due insiemi sono uguali se hanno gli stessi elementi. Il codice che ho scritto per la decomposizione conferma i risultati 

```
exams/2022-01-18B.pdf
Keys: {{'A', 'E'}, {'B', 'E'}, {'D', 'E'}}
Is 3NF? false
A decomposition is: {{'A', 'B', 'E'}, {'A', 'D'}, {'A', 'D', 'E'}, {'B', 'C'}, {'B', 'F'}, {'C', 'D'}}
```

**Reply by Alegau03** [reply]
> Perché dentro la decomposizione ho già AEB

Ah quindi se ho gli stessi elementi dentro la decomposizione posizione posso togliere

**Reply by niccolo-fato** [reply]
sì, nel mio caso avevo AEB e BEA, che sono uguali, quindi ho tolto uno dei due.

**Reply by Alegau03** [reply]
> sì, nel mio caso avevo AEB e BEA, che sono uguali, quindi ho tolto uno dei due.

Grazie ho capito!

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/636e773b-0f5f-4ed6-9285-7876f66d51e2)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/971534fe-4617-466d-aab2-4b34842f801a)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/6b96b40a-396d-4633-9881-78c67ad8ceff)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/c66190d0-0bc8-434d-a7ff-05aa8507a37b)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/84e1e1bf-b5c0-41d4-8195-dbbc89c08d58)
